src/Database/LSMTree/Internal.hs

```diff
@@ -1209,9 +1210,8 @@ createSnapshot snap label tableType t = do
 
         -- Guard that the snapshot does not exist already
         let snapDir = Paths.namedSnapshotDir (tableSessionRoot tEnv) snap
-        doesSnapshotExist <-
-          FS.doesDirectoryExist (tableHasFS tEnv) (Paths.getNamedSnapshotDir snapDir)
-        if doesSnapshotExist then
+        snapshotExists <- doesSnapshotExist (tableSession t) snap
```

Comment:
I like the reuse, but it's not great that it takes a second read lock on the session, though it does not matter much for performance or correctness reasons. It's just a minor nit that we don't need to take the lock twice. We can extract a function on `SessionEnv` (or just the components that it needs) that we then reuse instead